net: lib: Support setting fragment size in FOTA downloader

jtguggedal opened the pull request:
This PR adds support for setting fragment size in the FOTA download library.

jtguggedal commented:
FYI @sigvartmh and @jhn-nordic I had to do some rework after your approvals as some other changes to download client's fragment handling went into NCS.

@lemrey Please have a look and see if you're okay with the changes.

jhn-nordic commented on samples/nrf9160/http_application_update/src/main.c:
```diff
@@ -89,7 +89,8 @@ static void app_dfu_transfer_start(struct k_work *unused)
 				     sec_tag,
-				     apn);
+				     apn,
+				     0);
```
I sugggest we add a define called something like  "USE_FRAGMENT_SIZE_DEFAULTS" or similar

rlubos reviewed (approved):
LGTM, but I'd also like to hear from @lemrey as the author of the library.

lemrey commented on include/net/download_client.h:
```diff
@@ -97,6 +97,10 @@ struct download_client_cfg {
+	/** Maximum fragment size to download. 0 indicates that Kconfigured
+	 *  values shall be used.
+	 */
+	size_t max_fragment_size;
```
The idea is that `struct download_client_cfg` contains download-specific configs that the user might change between downloads, while `struct download_client` contains the lib's "private" state. In this case, it is appropriate to have the custom fragment size in `struct download_client_cfg` since the user might change that between downloads. If possible please rename to `frag_size_override`.

lemrey commented on include/net/fota_download.h:
```diff
@@ -78,13 +78,18 @@ int fota_download_init(fota_download_callback_t client_callback);
  * @param sec_tag Security tag you want to use with HTTPS set to -1 to Disable.
  * @param apn Access Point Name to use or NULL to use the default APN.
+ * @param fragment_size Fragment size to be used for the download.
+ *			If 0, CONFIG_DOWNLOAD_CLIENT_MAX_TLS_FRAGMENT_SIZE
+ *			is used for TLS connection and
+ *			CONFIG_DOWNLOAD_CLIENT_MAX_FRAGMENT_SIZE for
```
These two Kconfig were merged with my last changes into `DOWNLOAD_CLIENT_HTTP_FRAG_SIZE`.

lemrey reviewed (commented):
Overall looks good, but we should add a check in `download_client_connect()` to make sure that the custom fragment size is not larger than the size of buffer.

jtguggedal commented:
@lemrey I've addressed your comments. Diff here: https://github.com/nrfconnect/sdk-nrf/compare/aee6abca70931246000161dac9e4bc4078bab54e..7e530c93df6c50bc3f3f9155ad5ccb91d7731c87

Fixed the doxygen comment, added a check for custom fragment size larger than buffer and renamed the struct member to `fra_size_override`.

lemrey reviewed (approved):
Thanks, let's have an entry in the changelog.

rlubos commented:
> Thanks, let's have an entry in the changelog.

I've updated the changelog.
